### ISSUE TITLE
Memory leak in rc.SetItem()

### ISSUE THREAD

**Issue by OrestZborowski-SIG** [rtosholdings/riptable#289]
```
import riptide_cpp as rc
import tracemalloc

tracemalloc.start(10)

N = 10_000

for _ in range(N):
    rc.SetItem(None, "foo", "bar")

snapshot = tracemalloc.take_snapshot()
top_stats = snapshot.statistics('traceback')

stat = top_stats[0]
print("%s memory blocks: %.1f KiB" % (stat.count, stat.size / 1024))
for line in stat.traceback.format():
    print(line)
```
emits
```
30000 memory blocks: 1914.1 KiB
  File "C:\attr-leak.py", line 9
    rc.SetItem(None, "foo", "bar")
```

**Comment by OrestZborowski-SIG** [comment]
The non-np array value type is converted to a new np array type, and then is never decref'ed.

**Comment by OrestZborowski-SIG** [comment]
May affect more cases than just this...

**Comment by staffantj** [comment]
Fix in riptide_cpp/2c1f858c7c2c9857e6f92073fe71597965c3b281
